controllers/agentcluster_controller.go

```diff
@@ -348,3 +434,46 @@ func (r *AgentClusterReconciler) ensureSecretLabel(ctx context.Context, name, na
 		r.Log.WithError(err).Warn("Failed labeling secret")
 	}
 }
+
+func (r *AgentClusterReconciler) handleDeletion(ctx context.Context, agentCluster *capiproviderv1.AgentCluster) error {
+	if paused := agentCluster.Annotations[clusterv1.PausedAnnotation]; paused == "true" {
+		// unset finalizer, remove owner from owned resources to orphan them and return
+		if err := r.orphanClusterDeployment(ctx, agentCluster); err != nil {
```

Comment:
Do we need to do this both here and in the regular reconcile loop?

Reply:
I have this as a precaution - to make sure the resources always get orphaned. In case it gets wiped but paused around the same time. Do you think it's unnecessary?